validator/validator_spawner.go

```diff
@@ -0,0 +1,254 @@
+	err := os.MkdirAll(outDirPath, 0755)
+	if err != nil {
+		return err
+	}
+
+	rootPathAssign := ""
+	if executable, err := os.Executable(); err == nil {
+		rootPathAssign = "ROOTPATH=\"" + filepath.Dir(executable) + "\"\n"
+	}
+	cmdFile, err := os.OpenFile(filepath.Join(outDirPath, "run-prover.sh"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		return err
+	}
+	defer cmdFile.Close()
+	_, err = cmdFile.WriteString("#!/bin/bash\n" +
+		fmt.Sprintf("# expected output: batch %d, postion %d, hash %s\n", expOut.Batch, expOut.PosInBatch, expOut.BlockHash) +
+		"MACHPATH=\"" + machConf.getMachinePath(moduleRoot) + "\"\n" +
+		rootPathAssign +
+		"if (( $# > 1 )); then\n" +
+		"	if [[ $1 == \"-m\" ]]; then\n" +
+		"		MACHPATH=$2\n" +
+		"		shift\n" +
+		"		shift\n" +
+		"	fi\n" +
+		"fi\n" +
+		"${ROOTPATH}/bin/prover ${MACHPATH}/" + machConf.ProverBinPath)
+	if err != nil {
+		return err
+	}
+
+	for _, module := range machConf.LibraryPaths {
+		_, err = cmdFile.WriteString(" -l " + "${MACHPATH}/" + module)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = cmdFile.WriteString(fmt.Sprintf(" --inbox-position %d --position-within-message %d --last-block-hash %s", input.StartState.Batch, input.StartState.PosInBatch, input.StartState.BlockHash))
+	if err != nil {
+		return err
+	}
+
+	sequencerFileName := fmt.Sprintf("sequencer_%d.bin", input.StartState.Batch)
+	err = os.WriteFile(filepath.Join(outDirPath, sequencerFileName), sequencerMsg, 0644)
+	if err != nil {
+		return err
+	}
+	_, err = cmdFile.WriteString(" --inbox " + sequencerFileName)
+	if err != nil {
+		return err
+	}
+
+	preimageFile, err := os.Create(filepath.Join(outDirPath, "preimages.bin"))
+	if err != nil {
+		return err
+	}
+	defer preimageFile.Close()
+	for _, data := range input.Preimages {
+		lenbytes := make([]byte, 8)
+		binary.LittleEndian.PutUint64(lenbytes, uint64(len(data)))
+		_, err := preimageFile.Write(lenbytes)
+		if err != nil {
+			return err
+		}
+		_, err = preimageFile.Write(data)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = cmdFile.WriteString(" --preimages preimages.bin")
+	if err != nil {
+		return err
+	}
+
+	if input.HasDelayedMsg {
+		_, err = cmdFile.WriteString(fmt.Sprintf(" --delayed-inbox-position %d", input.DelayedMsgNr))
+		if err != nil {
+			return err
+		}
+		filename := fmt.Sprintf("delayed_%d.bin", input.DelayedMsgNr)
+		err = os.WriteFile(filepath.Join(outDirPath, filename), input.DelayedMsg, 0644)
+		if err != nil {
+			return err
+		}
+		_, err = cmdFile.WriteString(fmt.Sprintf(" --delayed-inbox %s", filename))
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = cmdFile.WriteString(" \"$@\"\n")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *ValidationSpawner) CreateExecutionBackend(ctx context.Context, wasmModuleRoot common.Hash, input *ValidationInput, targetMachineNum int) (*ExecutionChallengeBackend, error) {
```

Comment:
moving ExecutionBackend to validator is basically one big TODO.
executionBackend ultimately belongs in staker, with proper API.. but that's going to require more work which will be done later.